tests/test_ble_client_edge_cases.py

```
@@ -1,29 +1,74 @@
+
+def _raise_wrapped_not_implemented(
+    awaitable: Awaitable[object],
+    timeout: float | None = None,
+) -> NoReturn:
+    """Simulate `_async_await` wrapping NotImplementedError in BLEError."""
+    _ = timeout
+    close_awaitable = getattr(awaitable, "close", None)
+    if callable(close_awaitable):
+        close_awaitable()
+    try:
+        raise NotImplementedError("backend unsupported")
```

Ruff (0.15.4)

[warning] Abstract `raise` to an inner function

(TRY301)